docs/did/asset-did-spec.md

```diff
@@ -0,0 +1,107 @@
+The `modified-caip-19-asset-id` is derived from a CAIP asset identifier as described in the [CAIP-19 Asset Type and Asset ID Specification][caip-19-spec], with the `/` symbol separating the chain identifier from the asset identifier being replaced with the `.` symbol, and the optional `/` symbol preceding the `asset-id` component being replaced by the `:` symbol.
+This is required to render the resulting string a valid URI according to the [DID Core Specification][did-core-spec].
+
+Chain and asset namespaces are defined with Improvement Proposals and are stored under the [CAIP repository][caip-repo].
+For instance, the namespaces for [EIP155][caip-3-spec] (Ethereum-based) and [BIP122][caip-4-spec] (Bitcoin-based) chains as well as for [SLIP44][caip-20-spec] (native fungible tokens), [ERC20][caip-21-spec] (smart contract based tokens), and [ERC721][caip-22-spec] (NFTs) assets have already been defined.
+
+## Method Operations
+
+### Create an Asset DID
+
+A new Asset DID creation is automatically performed whenever a new asset class is deployed or a new asset instance for a given class minted.
+Because of the chain agnostic nature of Asset DIDs, this process can take place on any blockchain that is identifiable by a [CAIP-2 Identifier][caip-2-spec].
+
+### Resolve an Asset DID
+
+The information required to resolve an Asset DID is the following:
+
+- The **chain** on which the asset live, e.g., the Ethereum mainnet
+- The **namespace** defining the asset, e.g., the ERC721 namespace
+- The **reference** to the asset class within the defined namespace, e.g., the smart contract address for an NFT collection
+- [OPTIONAL] The unique **identifier** of the asset instance within the asset class, e.g., the NFT ID within a given collection
+
+The resolution process then follows a hierarchical structure.
+
+The chain identifier is decomposed into its namespace and reference.
+Then, the resolver for the specified namespace is used to resolve the specified reference.
+
+The same process is followed for the asset identifier.
+The identifier is decomposed into its namespace, reference, and optional ID.
+Then, the resolver for the specified asset namespace is used to resolve the asset reference and ID.
+
+As an example, the returned DID Document for [one of the CryptoKitties NFTs](https://opensea.io/assets/ethereum/0x06012c8cf97bead5deae237070f9587f8e7a266d/634446), identified with the Asset DID `did:asset:eip155:0.erc721:0x06012c8cf97bead5deae237070f9587f8e7a266d:634446`, is the following:
+
+```json
+{
+  "@context": "https://www.w3.org/ns/did/v1",
+  "id": "did:asset:eip155:0.erc721:0x06012c8cf97bead5deae237070f9587f8e7a266d:634446",
+  "controller": "did:pkh:eip155:0:0x929C3dAF2E2Be4C74A56EC01dF374bc46A34C6A1",
+}
+```
+
+where the `controller` field is optional and present only if the referenced asset class supports the concepts of ownership (which is not present, for instance, in the definition of a fungible asset class).
```

Comment:
Why not? This is all done at resolution time (by the resolver), and most NFT standards have a concept of ownership. I am not sure I fully understand your question.

Reply:
How I understood asset DIDs, it is not necessary to interact with the actual asset at all. I.e. resolving an asset DID just looks information up on the KILT chain and NOT on any other chain.

Reply:
This is also still my biggest gripe: what is the resolver architecture required to be able to do? This specification mentions checking for the existence of an asset and talks about chain & asset specific resolvers, which sounds like a massive piece of architecture. Do we really want to connect to all these chains? Or do we simply translate the CAIP-19, which is a process that would need no connection whatsoever? Timo keeps mentioning the KILT blockchain; is it planned to play any role in the resolution process and if so, which one?

Reply:
Alright, I see your point guys, and I am convinced that, for at least v1, we would not try to connect to all chains to fetch the owner information. I will change that. But I see very little point in having a DID specification that simply returns the DID that was queried and nothing more. It does not justify having a method on its own, and CAIP identifiers would suffice. Nevertheless, we have called it AssetDID, so I guess it would have to be a DID.

Would returning the different public credentials issued to the DID as service endpoints be something we want to put in? It would only require (minimal) interaction with the KILT blockchain, and clients would then process the returned value on their side, while we would just give instructions in the method itself.

Reply:
Makes sense to me, otherwise, as you said, it's just a DID shell for CAIP identifiers, which may not be terribly useful. If we say that we introduced this DIDification so we could have a way of listing public credentials associated with that identifier, I imagine people could accept that. In fact, if the asset DIDs don't add any features to CAIP identifiers, I could see that people ask why we not simply use CAIP identifiers as the credential subject.